packages/select/src/common/listItemsProps.ts

```diff
@@ -34,6 +34,13 @@ export interface IListItemsProps<T> extends IProps {
     /** Array of items in the list. */
     items: T[];
 
+    /**
+     * Whether the provided item fully matches the provided value. This is used
+     * to match pasted values to existing items, so that items can be emitted
+     * via `onItemsPaste`.
+     */
+    itemEqualsQuery?: (item: T, query: string) => boolean;
```

Comment:
isn't this just `itemPredicate`? https://github.com/palantir/blueprint/blob/develop/packages/select/src/common/predicate.ts#L23

i don't see the need for a new prop here, except that it slightly changes the semantics of `itemPredicate` in that you may want to define `itemListPredicate` for fast querying _and_ `itemPredicate` for pasting, instead of one-or-the-other.

Reply:
@giladgray They're unfortunately different:
- `itemPredicate` is for filtering: an item matches if the query matches any substring of the item title (e.g.)
- `itemEqualsQuery` is for exact matching: an item matches iff the query === the full item title (e.g.) 

Possible to add an `exactMatch` flag or something in `ItemPredicate<T>`?